Userland/Libraries/LibCore/System.cpp

```diff
@@ -1315,6 +1315,32 @@ ErrorOr<ssize_t> recvfrom(int sockfd, void* buffer, size_t buffer_length, int fl
     return received;
 }
 
+ErrorOr<AddressInfoVector> getaddrinfo(char const* nodename, char const* servname, struct addrinfo const& hints)
+{
+    struct addrinfo* results = nullptr;
+
+    int const rc = ::getaddrinfo(nodename, servname, &hints, &results);
+    if (rc != 0) {
+        if (rc == EAI_SYSTEM) {
+            return Error::from_syscall("getaddrinfo"sv, -errno);
+        }
+
+        auto const* error_string = gai_strerror(rc);
+        return Error::from_string_view({ error_string, strlen(error_string) });
+    }
+
+    Vector<struct addrinfo> addresses;
+
+    for (auto* result = results; result != nullptr; result = result->ai_next)
+        addresses.append(*result);
```

Comment:
Sorry, didn't catch this earlier. If we're adding fallible system calls, let's falliblize (yes that's a word) all the things:

```c++
for (auto* result = results; result != nullptr; result = result->ai_next)
    TRY(addresses.try_append(*result));
```

Reply:
Me when people are sorry for reviews multiple passes: That still only counts as one!